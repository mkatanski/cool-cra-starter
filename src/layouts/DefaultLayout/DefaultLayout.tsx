import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

import LayoutBase, { LayoutBaseProps } from "layouts/LayoutBase";
import { LinkButton } from "components/LinkButton";
import { TopBar } from "components/TopBar";
import { paths } from "config/paths";

const PageContent = styled.div`
  padding: 20px;
  height: 100%;
`;

const DefaultLayout: React.FC<LayoutBaseProps> = ({
  children,
  ...rest
}: LayoutBaseProps) => (
  <>
    <TopBar>
      <TopBar.Container>
        <Link to={paths.dashboard}>
          <LinkButton>Home</LinkButton>
        </Link>
      </TopBar.Container>
      <TopBar.Container>
        <Link to={paths.empty}>
          <LinkButton>Page 1</LinkButton>
        </Link>
      </TopBar.Container>
      <TopBar.Container>
        <Link to={paths.empty}>
          <LinkButton>Page 2</LinkButton>
        </Link>
      </TopBar.Container>
      <TopBar.Container alignRight>
        <LinkButton>Switch Theme</LinkButton>
      </TopBar.Container>
    </TopBar>
    <PageContent>
      <LayoutBase {...rest}>{children}</LayoutBase>
      <Link to={paths.login}>
        <LinkButton>Login view</LinkButton>
      </Link>
    </PageContent>
  </>
);

export default DefaultLayout;
