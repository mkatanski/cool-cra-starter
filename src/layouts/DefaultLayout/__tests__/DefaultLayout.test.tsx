/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";

import DefaultLayout from "../DefaultLayout";

jest.mock(
  "layouts/LayoutBase",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="LayoutBase" {...rest}>
      {children}
    </div>
  )
);

jest.mock("components/TopBar", () => {
  const TopBar = ({ children }: React.PropsWithChildren<{}>) => (
    <div id="TopBar">{children}</div>
  );
  TopBar.Container = ({ children }: React.PropsWithChildren<{}>) => (
    <div id="TopBar.Container">{children}</div>
  );

  return {
    TopBar,
  };
});

jest.mock("components/LinkButton", () => ({
  LinkButton: ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="LinkButton" {...rest}>
      {children}
    </div>
  ),
}));

describe("layouts", () => {
  describe("DefaultLayout", () => {
    it("should render DefaultLayout with children", () => {
      const tree = TestRenderer.create(
        <DefaultLayout>Some children</DefaultLayout>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
