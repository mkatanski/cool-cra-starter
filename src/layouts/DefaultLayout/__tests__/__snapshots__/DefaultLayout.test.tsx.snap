// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts DefaultLayout should render DefaultLayout with children 1`] = `
Array [
  <div
    id="TopBar"
  >
    <div
      id="TopBar.Container"
    >
      <div
        id="Link"
        to="/dashboard"
      >
        <div
          id="LinkButton"
        >
          Home
        </div>
      </div>
    </div>
    <div
      id="TopBar.Container"
    >
      <div
        id="Link"
        to="/empty"
      >
        <div
          id="LinkButton"
        >
          Page 1
        </div>
      </div>
    </div>
    <div
      id="TopBar.Container"
    >
      <div
        id="Link"
        to="/empty"
      >
        <div
          id="LinkButton"
        >
          Page 2
        </div>
      </div>
    </div>
    <div
      id="TopBar.Container"
    >
      <div
        id="LinkButton"
      >
        Switch Theme
      </div>
    </div>
  </div>,
  .c0 {
  padding: 20px;
  height: 100%;
}

<div
    className="c0"
  >
    <div
      id="LayoutBase"
    >
      Some children
    </div>
    <div
      id="Link"
      to="/login"
    >
      <div
        id="LinkButton"
      >
        Login view
      </div>
    </div>
  </div>,
]
`;
